patch from upstream

https://bugs.gentoo.org/426424

From ce92e974b8ecff19cf430b7b58a09b8190645b5c Mon Sep 17 00:00:00 2001
From: David Howells <dhowells@redhat.com>
Date: Fri, 26 Feb 2016 10:07:00 +0000
Subject: [PATCH] TEST: Fix endianness determination

Endianness determination was broken sometime after Fedora 20 when
executables switched to being DYN objects rather than EXEC objects.  This
caused the output of file to change.

Signed-off-by: David Howells <dhowells@redhat.com>
---
 tests/toolbox.inc.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/toolbox.inc.sh b/tests/toolbox.inc.sh
index f2463c57b40e..5ac23ffc04fb 100644
--- a/tests/toolbox.inc.sh
+++ b/tests/toolbox.inc.sh
@@ -13,10 +13,10 @@
 echo === $OUTPUTFILE ===
 
 endian=`file -L /proc/$$/exe`
-if expr "$endian" : '.* MSB \+executable.*' >&/dev/null
+if expr "$endian" : '.* MSB \+\(executable\|shared object\).*' >&/dev/null
 then
     endian=BE
-elif expr "$endian" : '.* LSB \+executable.*' >&/dev/null
+elif expr "$endian" : '.* LSB \+\(executable\|shared object\).*' >&/dev/null
 then
     endian=LE
 else
-- 
2.11.0

